File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
from collections import deque
n,m,v=map(int,input().split())
#인접행렬생성
graph=[[False]*(n+1) for _ in range(n+1)]
for _ in range(m):
    x,y=map(int,input().split())
    graph[x][y]=1
    graph[y][x]=1

##dfs
vis1=[False]*(n+1) #dfs 방문확인
def dfs(v):
    vis1[v]=True #방문
    print(v,end=' ') #방문한 곳 출력
    for i in range(1,n+1):
        if not vis1[i] and graph[v][i]==1:
            dfs(i)

##bfs
vis2=[False]*(n+1)
def bfs(v):
    q=deque([v])
    vis2[v]=True
    while q:
        v=q.popleft()
        print(v,end=' ')
        for i in range(1,n+1):
            if not vis2[i] and graph[v][i]==1:
                q.append(i)
                vis2[i]=True

dfs(v)
print()
bfs(v)

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

public class Main {

    static int N, M, V;
    static List<Integer>[] graph;
    static boolean[] visited;
    static StringBuilder sb;

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        N = Integer.parseInt(st.nextToken());
        M = Integer.parseInt(st.nextToken());
        V = Integer.parseInt(st.nextToken());

        graph = new ArrayList[N + 1];
        for (int i = 0; i < N + 1; i++) {
            graph[i] = new ArrayList<>();
        }

        for (int i = 0; i < M; i++) {
            st = new StringTokenizer(br.readLine());
            int a = Integer.parseInt(st.nextToken());
            int b = Integer.parseInt(st.nextToken());

            graph[a].add(b);
            graph[b].add(a);
        }

        for (int i = 1; i <= N; i++) {
            Collections.sort(graph[i]);
        }

        visited = new boolean[N + 1];
        sb = new StringBuilder();
        dfs(V);

        sb.append("\n");

        visited = new boolean[N + 1];
        bfs(V);

        System.out.println(sb);
    }

    private static void dfs(int v) {
        visited[v] = true;
        sb.append(v + " ");

        for (int i : graph[v]) {
            if (!visited[i]) {
                dfs(i);
            }
        }
    }

    private static void bfs(int v) {
        Queue<Integer> q = new ArrayDeque<>();
        q.offer(v);
        visited[v] = true;

        while (!q.isEmpty()) {
            int current = q.poll();

            sb.append(current + " ");

            for (int i : graph[current]) {
                if (!visited[i]) {
                    q.offer(i);
                    visited[i] = true;
                }
            }
        }
    }
}